calculator/operations.py

```diff
@@ -1,23 +1,37 @@
+from functools import reduce
+import operator
+
 
 def add(*args):
-    # your implementation here
-    pass
+    return sum(args)
 
 def subtract(*args):
-    # your implementation here
-    pass
+    if args:
+        answ = args[0]
+        for item in args[1:]:
+            answ = answ-item
+        return answ
+    else:
+        raise ArithmeticError
 
 def multiply(*args):
-    # your implementation here
-    pass
+    mult = reduce(operator.mul, args, 1)
+    return mult
 
 def divide(*args):
-    # your implementation here
-    pass
+    if args:
+        answ = float(args[0])
+        for item in args[1:]:
+            answ = answ/float(item)
+        return answ
+    else:
+        raise ArithmeticError
 
 def plot(*args):
     # OPTIONAL EXTRA CREDIT FUNCTION! 
     # See README for info.
     pass
 
 # add your custom operations here
+
+#PYTHONPATH=. py.test tests/test_main.py
```

Comment:
Well done! I liked how concise the calculator creator function was. 

The only thing: I don't think you have to return anything in your reset history function. Setting history to empty was enough